state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1776,7 +1872,57 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		let deposit_channel_details = DepositChannelLookup::<T, I>::get(&deposit_address)
 			.ok_or(Error::<T, I>::InvalidDepositAddress)?;
 
+		ensure!(
+			deposit_channel_details.deposit_channel.asset == asset,
+			Error::<T, I>::AssetMismatch
+		);
+
 		let channel_id = deposit_channel_details.deposit_channel.channel_id;
+		let channel_owner = deposit_channel_details.clone().owner;
+
+		if let Some(tainted_tx) =
+			TaintedTransactions::<T, I>::take(channel_owner.clone(), deposit_details.clone())
+		{
+			if tainted_tx.broker == channel_owner {
```

Comment:
Since we are reading this from a storage item with a key of `channel_owner`, this condition should always be true (see my comment about not really needed the `broker` field).

Reply:
Good point. This is redundant since we changed from `TxId->TaintedDetails` to `AccountId->TxId->TaintedDetails`